run-pre-checks.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+readonly script_directory="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+readonly project_directory="$(readlink -f "${script_directory}")"
+
+# shellcheck source=metadata.sh
+source "${project_directory}/metadata.sh"
+
+FAILED=0
+
+cd "${project_directory}"
+./tools/compile_templates.sh --check-only
+FAILED=${FAILED-$?}
```

Comment:
This is not even reachable on failure with `set -e`. Please delete.

Reply:
tbh, I'd rather remove the "-e" and handle exit codes manually. This would allow us to have multiple checks that are all executed, regardless if one of them fails...

Reply:
That can still be done with an `if`.